lazy_theta_star_b/include/lazy_theta_star_b/lazy_theta_star_b.h

```diff
@@ -0,0 +1,176 @@
+/*********************************************************************
+ *
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2020 Anshumaan Singh
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of Willow Garage, Inc. nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *
+ * Author: Anshumaan Singh
+ *********************************************************************/
+#include<cmath>
+#include<iostream>
+#include<cstdlib>
+#include<cstring>
+#include <string>
+#include<queue>
+#include<algorithm>
+
+#include "rclcpp/rclcpp.hpp"
+
+#include "nav2_core/global_planner.hpp"
+#include "nav_msgs/msg/path.hpp"
+#include "nav2_util/robot_utils.hpp"
+#include "nav2_util/lifecycle_node.hpp"
+#include "nav2_costmap_2d/costmap_2d_ros.hpp"
+#include "nav2_costmap_2d/cost_values.hpp"
+
+namespace lazy_theta_star_b {
+    typedef int id;
+    using namespace std;
+
+#define SMALL 0.001
+#define INF_COST 10000000.0
+#define LETHAL_COST 100
+
+    template<typename ptsT>
+    struct pts {
+        ptsT x, y;
+    };
+
+    struct pos {
+        id pos_id;
+        double *g;
+        double *f;
+    };
+
+    struct tree_node {
+        int x, y;
+        double g = INF_COST;
+        double h = INF_COST;
+        id parentId;
+        int closed = 0;       //0 - unexplored, 1 - closed, -1 - open
+        double f = INF_COST;
+        int counter = 0;
+        int got_here = 0;
+    };
+
+
+    class LazyThetaStarB : public nav2_core::GlobalPlanner {
```

Comment:
Typically we try to isolate the algorithm from the ROS wrapper. So typically you have a `lazy_theta_star` planner algorithm that takes in a generic cost field, starting / goal poses, and does the planning. Then the plugin will have that planner algorithm as an object in the class that the ROS wrapped plugin will call. It helps decouple the ROS stuff from the algorithm stuff so that it can be more easily ported and tested.

Reply:
Yeah, I have a version, that has done exactly that, will commit it along with the changes as mentioned below.